shared/pingpong/accounts.go

```diff
@@ -186,133 +194,156 @@ func prepareAssets(accounts map[string]uint64, client libgoal.Client, cfg PpConf
 			totalSent++
 			throttleTransactionRate(startTime, cfg, totalSent)
 		}
-		account, accountErr := client.AccountInformation(addr)
-		if accountErr != nil {
-			fmt.Printf("Cannot lookup source account %v\n", addr)
-			err = accountErr
-			return
-		}
+	}
 
+	// wait until all the assets created
+	allAssets := make(map[uint64]string, cfg.NumAsset)
+	for addr := range accounts {
+		var account v1.Account
+		start := time.Now()
+		for {
+			account, err = client.AccountInformation(addr)
+			if err != nil {
+				fmt.Printf("Warning, cannot lookup source account")
+				return
+			}
+			if len(account.AssetParams) >= numCreatedAssetsByAddr[addr] {
+				break
+			}
+			if time.Now().After(start.Add(3 * time.Minute)) {
```

Comment:
create a `deadline = start.Add(3 * time.Minute)` outside the for loop and test against it.

Reply:
done